jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/KModuleDeploymentServiceTest.java

```diff
@@ -254,7 +253,61 @@ public void testUnDeploymentWithActiveProcesses() {
 
         engine.getKieSession().abortProcessInstance(processInstance.getId());
 
-        checkFormsDeployment( deploymentUnit.getIdentifier() );
+        checkFormsDeployment(deploymentUnit.getIdentifier());
+    }
+
+    @Test
+    public void testUnDeploymentWithoutActiveProcesses() {
+
+        assertNotNull(deploymentService);
+
+        DeploymentUnit deploymentUnit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
+        deploymentService.deploy(deploymentUnit);
+        units.add(deploymentUnit);
+        DeployedUnit deployedGeneral = deploymentService.getDeployedUnit(deploymentUnit.getIdentifier());
+
+        assertNotNull(deployedGeneral);
+        assertNotNull(deployedGeneral.getDeploymentUnit());
+        assertNotNull(deployedGeneral.getRuntimeManager());
+
+        RuntimeManager manager = deploymentService.getRuntimeManager(deploymentUnit.getIdentifier());
+        assertNotNull(manager);
+
+        RuntimeEngine engine = manager.getRuntimeEngine(EmptyContext.get());
+        assertNotNull(engine);
+
+        deploymentService.undeploy(deploymentUnit);
+    }
+
+    @Test
+    public void testUnDeploymentWithActiveProcessesAborting() {
+        assertNotNull(deploymentService);
+
+        DeploymentUnit deploymentUnit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
+        deploymentService.deploy(deploymentUnit);
+        units.add(deploymentUnit);
+        DeployedUnit deployedGeneral = deploymentService.getDeployedUnit(deploymentUnit.getIdentifier());
+        assertNotNull(deployedGeneral);
+        assertNotNull(deployedGeneral.getDeploymentUnit());
+        assertNotNull(deployedGeneral.getRuntimeManager());
+
+        RuntimeManager manager = deploymentService.getRuntimeManager(deploymentUnit.getIdentifier());
+        assertNotNull(manager);
+
+        RuntimeEngine engine = manager.getRuntimeEngine(EmptyContext.get());
+        assertNotNull(engine);
```

Comment:
please dispose the engine once done with processes

Reply:
@mswiderski done, I also disposed it on other tests...